notebooks/03-query/11-joins.ipynb

```diff
@@ -0,0 +1,2475 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "id": "20204185-47af-4f1f-a965-9a759a0703a9",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{\n",
+      "  \"name\": \"clicks\",\n",
+      "  \"config_file\": \"clickstream/clickstream.json\",\n",
+      "  \"target\": {\n",
+      "    \"type\": \"file\",\n",
+      "    \"path\": \"/files/clicks.json\"\n",
+      "  },\n",
+      "  \"active_sessions\": 0,\n",
+      "  \"total_records\": 1208839,\n",
+      "  \"start_time\": \"2023-10-12 00:00:00\",\n",
+      "  \"run_time\": 1209666.263514,\n",
+      "  \"status\": \"COMPLETE\",\n",
+      "  \"status_msg\": \"Running, Sim Clock: 2023-10-26 00:01:06.264514\"\n",
+      "}\n",
+      "2023-10-26 01:11:27.949649 <<<<<<<<<<<<< after sales data generation\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "datetime.datetime(2023, 10, 26, 1, 11, 27, 949649)"
+      ]
+     },
+     "execution_count": 29,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# simulate forecast for last 2 weeks \n",
+    "gen_hours=24*14  \n",
+    "gen_now = datetime.now().replace(hour=0,minute=0,second=0) - timedelta(hours=gen_hours)\n",
+    "gen_start_time = gen_now.strftime(\"%Y-%m-%d %H:%M:%S\")\n",
+    "\n",
+    "headers = {\n",
+    "  'Content-Type': 'application/json'\n",
+    "}\n",
+    "\n",
+    "datagen_request = {\n",
+    "    \"name\": \"forecast\",\n",
+    "    \"target\": { \"type\": \"file\", \"path\":\"forecast.json\"  },\n",
+    "    \"config_file\": \"clickstream/sales_forecast.json\", \n",
+    "    \"time\": f\"{gen_hours}h\",\n",
+    "    \"concurrency\":1040,\n",
+    "    \"time_type\":gen_start_time\n",
+    "}\n",
+    "datagen.post(\"/start\", json.dumps(datagen_request), headers=headers)\n",
+    "wait_for_datagen(\"forecast\")\n",
+    "print_timestamp(\"after forecast data generation\")\n",
+    "\n",
+    "datagen_request = {\n",
+    "    \"name\": \"clicks\",\n",
+    "    \"target\": { \"type\": \"file\", \"path\":\"clicks.json\"  },\n",
+    "    \"config_file\": \"clickstream/clickstream.json\", \n",
+    "    \"time\": f\"{gen_hours}h\",\n",
+    "    \"concurrency\":20,\n",
+    "    \"time_type\":gen_start_time\n",
+    "}\n",
+    "datagen.post(\"/start\", json.dumps(datagen_request), headers=headers)\n",
+    "wait_for_datagen(\"clicks\")\n",
+    "print_timestamp(\"after sales data generation\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5a76c614-556a-427f-8398-cb9e1eeb6254",
+   "metadata": {},
+   "source": [
+    "### Natural Ingestion\n",
+    "For the first two query patterns, we ingest the data mostly \"as is\". \n",
```

Comment:
Make this active:

Run the following cells to ingest the data.

Reply:
done